lingua_franca/lang/format_en.py

```diff
@@ -302,6 +304,43 @@ def _long_scale(n):
     return result
 
 
+def pronounce_digits_en(number, places=2, all_digits=False):
+    decimal_part = ""
+    op_val = ""
+    result = []
+    is_float = isinstance(number, float)
+    if is_float:
+        op_val, decimal_part = [part for part in str(number).split(".")]
+        decimal_part = pronounce_number_en(
+            float("." + decimal_part), places=places).replace("zero ", "")
+    else:
+        op_val = str(number)
+
+    if all_digits:
+        result = [pronounce_number_en(int(i)) for i in op_val]
+        if is_float:
+            result.append(decimal_part)
+        result = " ".join(result)
+    else:
+        while len(op_val) > 1:
+            idx = -2 if len(op_val) in [2, 4] else -3
+            back_digits = op_val[idx:]
+            op_val = op_val[:idx]
+            result = pronounce_number_en(
+                int(back_digits)).split(" ") + result
+        if op_val:
+            result.insert(0, pronounce_number_en(int(op_val)))
+        if is_float:
+            result.append(decimal_part)
+        no_no_words = list(_SHORT_SCALE_EN.values())[:5]
+        no_no_words.append('and')
+        print(no_no_words)
+        print(result)
+        result = [word for word in result if word.strip() not in no_no_words]
```

Comment:
Is there any case where you think this might happen that we can test for? Or is it just a safety measure?

Reply:
This happens anytime the input is longer than two digits. The algorithm acts by running `pronounce_number()` on 2-3 digits at a time. This often returns the words `hundred` and `and`.

The latter stray debug print (=P) is the result prior to this operation:

```python
>>> pronounce_digits(234534)
['two', 'hundred', 'and', 'thirty', 'four', 'five', 'hundred', 'and', 'thirty', 'four']
'two thirty four five thirty four'
```

`pronounce_number(534)`, prepended with `pronounce_number(234)`, sanitized.

Reply:
The strip, on the other hand, is probably unneeded.